src/spikeinterface/widgets/unit_waveforms.py

```diff
@@ -106,18 +106,25 @@ def __init__(
         sorting: BaseSorting = we.sorting
 
         if unit_ids is None:
-            unit_ids = sorting.get_unit_ids()
-        unit_ids = unit_ids
+            unit_ids = sorting.unit_ids
         if channel_ids is None:
             channel_ids = we.channel_ids
-
         if unit_colors is None:
             unit_colors = get_unit_colors(sorting)
 
         channel_locations = we.get_channel_locations()[we.channel_ids_to_indices(channel_ids)]
 
+        extra_sparsity = False
         if waveform_extractor.is_sparse():
-            sparsity = waveform_extractor.sparsity
+            if sparsity is None:
+                sparsity = waveform_extractor.sparsity
+            else:
+                # assert provided sparsity is sparser than the one in the waveform extractor
+                waveform_sparsity = we.sparsity
+                assert np.all(
+                    np.sum(waveform_sparsity.mask, 1) - np.sum(sparsity.mask, 1) > 0
+                ), "The provided sparsity needs to be sparser than the one in the waveform extractor!"
```

Comment:
Is there any way we can help the user to understand what they did wrong? I can't think of a good way to do this to help the user fix their mistake, but it might be nice if we could provide more info. I love this sortingview summary, so I think more people are eventually going to use it.

Reply:
I'll give it a try to be more explicative in the assertion message

Reply:
Yeah that’s great!